### ISSUE TITLE
Regression on parsing 

### ISSUE THREAD

**Issue by casidiablo** [nushell/nushell#1987]
**Describe the bug**

This used to work:

```
cat some.json | jq -r "sort_by(.monthly_cost)"
```

Now I get:

```
sh: -c: line 0: syntax error near unexpected token `('
sh: -c: line 0: `jq -r sort_by(.monthly_cost)'
```

**Expected behavior**

It should work

**Configuration (please complete the following information):**
 - OS (e.g. Windows): NixOS
 - Nu version: 0.15.0

**Comment by jonathandturner** [comment]
I suspect the `(` needs to be properly escaped before we call out to `jq`. If that's what it is, the fix shouldn't be too bad.

I'll flag this as "Good First Issue". If someone picks this up and turns out it's more complicated, just tag me and we'll figure out what it is.

**Comment by bailey-layzer** [comment]
@jonathandturner I'm new to this project (and Rust) and took a run at this. It looks like jq expects the argument quoted, but the quotes (single or double) are stripped off by the new parser here:
https://github.com/nushell/nushell/blob/831111edec221fddb2549f079170c3451e06a8cb/crates/nu-parser/src/parse.rs#L565
I "fixed" this by just cloning `lite_arg` into `trimmed` and commenting out this line (the new `\` causes another syntax error):			https://github.com/nushell/nushell/blob/831111edec221fddb2549f079170c3451e06a8cb/crates/nu-cli/src/commands/classified/external.rs#L135
I'm guessing you'll want to deal with this another way - thoughts?

**Comment by jonathandturner** [comment]
@bailey-layzer - sorry, didn't see this comment earlier.

So you're saying that by us making sure the output has quotes in it, we're causing the issue? Interesting. When I first read it, it seemed like we weren't adding quotes when we should be.

We used to have different logic for when quotes where automatically added to arguments for external commands. I wonder if we should make that logic a little smarter.

**Comment by bailey-layzer** [comment]
@jonathandturner the quotes are added back to external command arguments containing a space:
`cat test.json | jq -r ".users | sort_by(.firstName)"` and `cat test.json | jq -r ".users|sort_by( .firstName)"` work, but
`cat test.json | jq -r ".users|sort_by(.firstName)` gives me
<pre>sh: -c: line 0: syntax error near unexpected token `.firstName&apos;
sh: -c: line 0: `jq -r .users|sort_by(.firstName)&apos;
</pre>

**Comment by bailey-layzer** [comment]
This issue was fixed by #2095.
